tests/Console/E2ETest.php

```diff
@@ -0,0 +1,243 @@
+            $this->markTestSkipped(implode("\n", [
+                'Refusing to run Infection on itself with no memory limit set: it is dangerous.',
+                'To run this test with a memory limit set please use:',
+                'php -dmemory_limit=128M vendor/bin/phpunit --group=large',
+            ]));
+        }
+
+        $output = $this->runInfection(self::EXPECT_SUCCESS, [
+            '--test-framework-options="--exclude-group=e2e"',
+        ]);
+
+        $this->assertRegExp('/\d+ mutations were generated/', $output);
+        $this->assertRegExp('/\d{2,} mutants were killed/', $output);
+    }
+
+    /**
+     * @group e2e
+     */
+    public function test_it_runs_configure_command_if_no_configuration()
+    {
+        chdir('tests/Fixtures/e2e/Unconfigured/');
+
+        $output = $this->runInfection(self::EXPECT_ERROR);
+
+        $this->assertContains(ConfigureCommand::NONINTERACTIVE_MODE_ERROR, $output);
+    }
+
+    /**
+     * @dataProvider e2eTestSuiteDataProvider
+     * @group e2e
+     */
+    public function test_it_runs_an_e2e_test_with_success(string $e2eTestName, string $fullPath)
+    {
+        $this->runOnE2EFixture($fullPath);
+    }
+
+    public function e2eTestSuiteDataProvider(): \Generator
+    {
+        $directories = Finder::create()
+            ->depth('== 0')
+            ->in(__DIR__ . '/../Fixtures/e2e/')
+            ->directories();
+
+        foreach ($directories as $dirName) {
+            if (file_exists($dirName . '/run_tests.bash')) {
+                // skipping non-standard tests
+                // specifically Memory_Limit - it is very slow to fail
+                continue;
+            }
+
+            yield [
+                // Makes PHPUnit report readable errors on failure:
+                // ...with data set #0 ('Foo_bar', ...
+                basename((string) $dirName),
+                $dirName,
+            ];
+        }
+    }
+
+    protected function tearDown()
+    {
+        if ($this->previousLoader) {
+            $this->previousLoader->unregister();
+        }
+
+        chdir($this->cwd);
+    }
+
+    private function runOnE2EFixture($path): string
+    {
+        $this->assertTrue(is_dir($path));
+        chdir($path);
+
+        $this->installComposerDeps();
+        $output = $this->runInfection(self::EXPECT_SUCCESS);
+
+        $this->assertRegExp('/You are running Infection with \w+ enabled./', $output);
+        $this->assertRegExp('/\d+ mutations were generated/', $output);
+        $this->assertRegExp('/\d+ mutants were killed/', $output);
+
+        if (isset($_SERVER['GOLDEN'])) {
+            copy('infection-log.txt', 'expected-output.txt');
+            $this->markTestSkipped('Saved golden output');
+        }
+
+        $this->assertFileEquals('expected-output.txt', 'infection-log.txt', sprintf('%s/expected-output.txt is not same as infection-log.txt (if that is OK, run GOLDEN=1 vendor/bin/phpunit)', getcwd()));
```

Comment:
Save some time copying files from here to where when adding a new mutator: running `GOLDEN=1 vendor/bin/phpunit` should be enough to cover most bases.

Reply:
could you please explain why `GOLDEN`? ;) does it have special meaning in other technical places, just curious?

Also, I would say we should add this info to https://github.com/infection/infection/blob/master/.github/CONTRIBUTING.md (we will move dev documentation to the separate file or event to `infection/site` but that's definitely out of the scope of this PR)

Reply:
I think `CONTRIBUTING.md` could be left alone in this case because if you would scroll to the right a failing test would say exactly what you'd want to do:

`$this->assertFileEquals('expected-output.txt', 'infection-log.txt', sprintf('%s/expected-output.txt is not same as infection-log.txt (if that is OK, run GOLDEN=1 vendor/bin/phpunit)', getcwd()));`

Reply:
thanks, no problem